elftools/ehabi/ehabiinfo.py

```diff
@@ -0,0 +1,212 @@
+        """
+        if self._num_entry is None:
+            self._num_entry = self._arm_idx_section.header['sh_size'] // EHABI_INDEX_ENTRY_SIZE
+        return self._num_entry
+
+    def get_entry(self, n):
+        """ Get the exception handler entry at index #n. (EHABIEntry object or a subclass)
+        """
+        if n >= self.num_entry():
+            raise IndexError('Invalid entry %d/%d' % (n, self._num_entry))
+        eh_index_entry_offset = self._arm_idx_section['sh_offset'] + n * EHABI_INDEX_ENTRY_SIZE
+        eh_index_data = struct_parse(self._struct.EH_index_struct, self._arm_idx_section.stream, eh_index_entry_offset)
+        Word0, Word1 = eh_index_data['Word0'], eh_index_data['Word1']
+
+        if Word0 & 0x80000000 != 0:
+            print ('Corrupt ARM exception handler table entry: %x' % n)
+            return CorruptEHABIEntry()
+
+        function_offset = arm_expand_prel31(Word0, self._arm_idx_section['sh_offset'] + n * EHABI_INDEX_ENTRY_SIZE)
+
+        if Word1 == 1:
+            # 0x1 means cannot unwind
+            return CannotUnwindEHABIEntry(function_offset)
+        elif Word1 & 0x80000000 == 0:
+            # highest bit is zero, point to .ARM.extab data
+            eh_table_offset = arm_expand_prel31(Word1, self._arm_idx_section['sh_offset'] + n * EHABI_INDEX_ENTRY_SIZE + 4)
+            eh_index_data = struct_parse(self._struct.EH_table_struct, self._arm_idx_section.stream, eh_table_offset)
+            Word0 = eh_index_data['Word0']
+            if Word0 & 0x80000000 == 0:
+                # highest bit is one, generic model
+                return GenericEHABIEntry(function_offset, arm_expand_prel31(Word0, eh_table_offset))
+            else:
+                # highest bit is one, arm compact model
+                # highest half must be 0b1000 for compact model
+                if Word0 & 0x70000000 != 0:
+                    print ('Corrupt ARM compact model table entry: %x' % n)
+                    return CorruptEHABIEntry()
+                per_index = (Word0 >> 24) & 0x7f
+                if per_index == 0:
+                    # arm compact model 0
+                    opcode = [(Word0 & 0xFF0000) >> 16, (Word0 & 0xFF00) >> 8, Word0 & 0xFF]
+                    return EHABIEntry(function_offset, per_index, opcode)
+                elif per_index == 1 or per_index == 2:
+                    # arm compact model 1/2
+                    more_word = (Word0 >> 16) & 0xff
+                    opcode = [(Word0 >> 8) & 0xff, (Word0 >> 0) & 0xff]
+                    self._arm_idx_section.stream.seek(eh_table_offset + 4)
+                    for i in range(more_word):
+                        r = struct_parse(self._struct.EH_table_struct, self._arm_idx_section.stream)['Word0']
+                        opcode.append((r >> 24) & 0xFF)
+                        opcode.append((r >> 16) & 0xFF)
+                        opcode.append((r >> 8) & 0xFF)
+                        opcode.append((r >> 0) & 0xFF)
```

Comment:
these patterns are generally problematic for bi-endianness.

Reply:
Your concern is unnecessary, I've handled endianness compatibility with code:

```
struct_parse(self._struct.EH_table_struct, self._arm_idx_section.stream)
```

The `opcode` variable is uint32 in correct endianness.

Reply:
Would you please:
1. add the C source file that produced the test binaries.
2. add binaries compiled with `-mbig-endian` to prove the code handles bi-endianness.

Reply:
> add the C source file that produced the test binaries.

Done, the C source is very simple. `arm_exidx_test.cpp`.

> add binaries compiled with -mbig-endian to prove the code handles bi-endianness.

I've tried to compile and produce a big-endian elf, but I failed. 

Try1:  I'm an ARM developer with android-ndk, when I compile from C to shared library or executable, link error occurs. I notice that android-ndk doesn't provide big-endian environment.
Try2: I use `apt install gcc-arm-linux-gnueabi` and cross compile. Link error also occurs.

However, I convince that my code is compatible and do that test: I compare the `bytecode_array` between `self.little_endian=True` and `self.little_endian=False`, they are invert mutually.

Reply:
Did you forget to `git add` the test `arm_exidx_test.cpp`?

Reply:
Indeed, the NDK doesn't distribute the C runtime objects necessary for linking a cross compiled executable targeting arm big endian.  Ok, that

Reply:
> Did you forget to git add the test arm_exidx_test.cpp?

Sorry. Attached now.

> Though you can skip linking by just compiling object files with -c.

You're right. But, this is the limitation we discussed before, current version doesn't support relocatable object file.